packages/eslint-plugin-svelte/tests/fixtures/rules/no-add-event-listener/invalid/test01-input.svelte

```diff
@@ -0,0 +1,31 @@
+<script>
+	const handler = (ev) => {
+		console.log(ev);
+	};
+
+	function onClick(event) {
+		const target = event.currentTarget;
+
+		const deepObj = {
+			deep: {
+				obj: {
+					target
+				}
+			}
+		};
+
+		target.addEventListener('focus', handler);
+		deepObj.deep.obj.target.addEventListener('focus', handler);
+	}
+
+	addEventListener('message', handler);
+	window.addEventListener('message', handler);
+	document.addEventListener('visibilitychange', handler);
+
+	// with a load of whitespace
+	window.addEventListener    ('message', handler);
+	// with a comment
+	window.addEventListener/* foo */('message', handler);
```

Comment:
Should we also include the TypeScript case like this?

```ts
(window as any).addEventListener ('message', handler);
(window.addEventListener as any)('message', handler);
```

Reply:
for this one, i added the test cases but this one isn't currently supported:

```ts
(window.addEventListener as any)('message', handler);
```

we would need logic to handle call expressions containing casts, which feels weird. maybe we should just ignore this case and not try handle it?

Reply:
I’m not sure how complicated this is, so I’ll try implementing it myself later.

Reply:
its not complicated, its just awkward that we'd need code which knows about casts

```ts
((window.addEventListener) as any) as any)('message', handler)
```

the callee is then a `TSAsExpression`

which means we'd need to traverse this node specifically stripping out/ignoring casts (because only casts create this situation)

i.e. you would do this flow:

```
const actualNode = findFirstNonCastNode(node);

// that actually does this
while (expression.type is TSAsExpression) {
  expression = expression.expression;
}
```

sure it'd make it work but it makes me feel like this for some reason: 😬